java-testing-course-work-main/src/test/java/com/skypro/simplebanking/controller/AccountControllerTest.java

```diff
@@ -0,0 +1,127 @@
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void getUserAccount_WhenAccountNotFound() throws Exception {
+        Long id = 0L;
+        mockMvc.perform(get("/account/{id}", id)
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Lila", "dodo")))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void depositToAccount_Test() throws Exception {
+        mockMvc.perform(post("/account/deposit/{id}", getAccountId("Dima"))
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Dima", "lili"))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(getBalanceChangeRequest(1000L).toString()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.amount").value(1001L));
+    }
+
+    @Test
+    public void depositToAccount_WhenAccountNotFound() throws Exception {
+        Long id = -1L;
+        mockMvc.perform(post("/account/deposit/{id}", id)
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Dima", "lili"))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(getBalanceChangeRequest(1L).toString()))
+                .andExpect(status().isNotFound());
+    }
+    @Test
+    public void depositToAccount_WhenWrongCurrencyExeption() throws Exception {
+        Long id = 0L;
+        mockMvc.perform(post("/account/deposit/{id}", id)
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Dima", "lili"))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(getAccountCurrency("EUR").toString()))
+                .andExpect(status().isNotFound());
+    }
+    @Test
+    public void depositToAccount_WhenInvalidAmountException() throws Exception {
+        Long id = -1L;
+        mockMvc.perform(post("/account/deposit/{id}", id)
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Dima", "lili"))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(getBalanceChangeRequest(-1L).toString()))
+                .andExpect(status().isBadRequest());
+    }
+    @Test
+    public void withdrawToAccount_WhenInsufficienFunds() throws Exception {
+        mockMvc.perform(post("/account/withdraw/{id}", 1)
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Ilya", "nook"))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(getBalanceChangeRequest(1000L).toString()))
+                .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    public void withdrawToAccount_WhenAccountNotFound() throws Exception {
+        Long id = 0L;
+        mockMvc.perform(post("/account/withdraw/{id}", id)
+                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Ilya", "nook"))
```

Comment:
```suggestion
                        .header(HttpHeaders.AUTHORIZATION, base64Encoded("Ilya", "lolo"))
```
иначе тест падает, потому что нет юзера с таким паролем